integ-test/src/test/java/org/opensearch/sql/ppl/ParseCommandIT.java

```diff
@@ -62,14 +111,38 @@ public void testParseCommandWithOtherRunTimeFields() throws IOException {
                 "source=%s | parse email '.+@(?<host>.+)' | "
                     + "eval eval_result=1 | fields host, eval_result",
                 TEST_INDEX_BANK));
-    verifyOrder(
-        result,
-        rows("pyrami.com", 1),
-        rows("netagy.com", 1),
-        rows("quility.com", 1),
-        rows("boink.com", 1),
-        rows("scentric.com", 1),
-        rows("filodyne.com", 1),
-        rows("quailcom.com", 1));
+
+    // Create the expected rows as List<Object[]>
+    List<Object[]> expectedRows =
+        new ArrayList<>(
+            List.of(
+                new Object[] {"pyrami.com", 1},
+                new Object[] {"netagy.com", 1},
+                new Object[] {"quility.com", 1},
+                new Object[] {"boink.com", 1},
+                new Object[] {"scentric.com", 1},
+                new Object[] {"filodyne.com", 1},
+                new Object[] {"quailcom.com", 1}));
+
+    // Actual rows from the response
+    JSONArray dataRows = result.getJSONArray("datarows");
+    List<Object[]> actualRows = new ArrayList<>();
+    for (int i = 0; i < dataRows.length(); i++) {
+      JSONArray row = dataRows.getJSONArray(i);
+      actualRows.add(new Object[] {row.getString(0), row.getInt(1)});
+    }
+
+    if (expectedRows.size() != actualRows.size()) {
+      Assert.fail("Row count is different " + expectedRows + " " + actualRows);
+    }
+
+    // Sort both expected and actual rows
+    expectedRows.sort((a, b) -> ((String) a[0]).compareTo((String) b[0]));
+    actualRows.sort((a, b) -> ((String) a[0]).compareTo((String) b[0]));
+
+    // Compare the sorted lists by iterating over elements and using assertArrayEquals
+    for (int i = 0; i < expectedRows.size(); i++) {
+      assertArrayEquals(expectedRows.get(i), actualRows.get(i));
+    }
```

Comment:
I see very similar patterns here. Can't we come up with a common method for verification?

Reply:
Sure, tried to extract the assertion and helper functions as much possible. Updated.